src/components/ChataTable/ChataTable.js

```diff
@@ -504,18 +510,33 @@ export default class ChataTable extends React.Component {
       // We only use header filters so we have to use the function below
       const headerFilters = this.ref?.tabulator?.getHeaderFilters()
 
-      if (!this.props.useInfiniteScroll) {
-        this.tableParams.filter = _cloneDeep(headerFilters)
-        this.props.onFilterCallback(headerFilters, rows)
-      }
+      if (!this.useInfiniteScroll) {
+        const tableParamsFormatted = formatTableParams(
+          { ...this.tableParams, filter: headerFilters },
+          this.props.columns,
+        )
+
+        // Apply filters to original data to get true count
+        const filteredData = this.originalQueryData.filter((row) => {
```

Comment:
This is a pretty expensive bit of logic, i would consider moving it to its own function and updating it. 

Right now, the code applies each filter by looping over every row and calling filterDataByColumn on a single row array. This repeats work unnecessarily, for each row/filter pair we look up the column index with find and create a temporary array just to pass into filterDataByColumn.

Instead, we can build a Map of columnId → columnIndex once before, then iterate through filters and apply filterDataByColumn to the entire list each time using .reduce(). This reduces the filtering process a lot and makes the logic easier to read.